File: app/src/main/java/com/gcoders/schoolinfo/schoolsfindersystem/view/SchoolMainActivity.java
package com.gcoders.schoolinfo.schoolsfindersystem.view;

import android.content.Intent;
import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import androidx.appcompat.widget.Toolbar;
import android.widget.Toast;

import com.gcoders.schoolinfo.schoolsfindersystem.R;
import com.gcoders.schoolinfo.schoolsfindersystem.SchoolApplication;
import com.gcoders.schoolinfo.schoolsfindersystem.model.SchoolListInfo;
import com.gcoders.schoolinfo.schoolsfindersystem.model.SchoolSATResultInfo;
import com.gcoders.schoolinfo.schoolsfindersystem.presenter.SchoolListPresenter;
import com.gcoders.schoolinfo.schoolsfindersystem.utils.SchoolUtils;
import com.gcoders.schoolinfo.schoolsfindersystem.view.adapter.SchoolListAdapter;

import org.apache.commons.lang3.StringUtils;

import java.util.ArrayList;

public class SchoolMainActivity extends AppCompatActivity implements SchoolListInfoViewContract.View {

    private RecyclerView recyclerview_schools_list;
    private SchoolListInfoViewContract.Presenter schoolListPresenter;
    private ArrayList<SchoolListInfo> schoolListInfoObjects;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        schoolListInfoObjects = SchoolApplication.getsInstance().getSchoolListInfoObjects();

        schoolListPresenter = new SchoolListPresenter(this);

        recyclerview_schools_list = findViewById(R.id.recyclerview_schools_list);
        recyclerview_schools_list.setLayoutManager(new LinearLayoutManager(this, RecyclerView.VERTICAL, false));
        schoolListReady(schoolListInfoObjects);
    }


    @Override
    public void schoolListReady(ArrayList<SchoolListInfo> schoolListInfoObjects) {
        SchoolListAdapter adapter = new SchoolListAdapter(schoolListInfoObjects, new SchoolListAdapter.SchoolInfoClickListener() {
            @Override
            public void onSchoolInfoClick(SchoolListInfo schoolListInfo) {
                schoolListPresenter.getSchoolSATResultInfo(schoolListInfo.getDbn());
            }
        });
        recyclerview_schools_list.setAdapter(adapter);
    }

    @Override
    public void schoolInfoClicked(ArrayList<SchoolSATResultInfo> schoolSATResultInfos) {

        if (null != schoolSATResultInfos && schoolSATResultInfos.size() > 0) {
            if (validateSATResults(schoolSATResultInfos)) {
                Intent intent = new Intent(this, SchoolSATscoreActivity.class);
                Bundle bundle = new Bundle();
                bundle.putParcelable("schoolsatscore", schoolSATResultInfos.get(0));
                intent.putExtras(bundle);
                startActivity(intent);
            } else {
                onFailure("Incorrect info received. Please try again later..!");
            }
        } else {
            onFailure("No Results found. Please try a different search..!");
        }
    }

    @Override
    public void onFailure(String errorMessage) {
        Toast.makeText(this, errorMessage, Toast.LENGTH_SHORT).show();
    }

    private boolean validateSATResults(ArrayList<SchoolSATResultInfo> schoolSATResultInfos) {
        SchoolSATResultInfo mSATResultInfo = schoolSATResultInfos.get(0);
        if (StringUtils.isNotBlank(mSATResultInfo.getDbn())
                && StringUtils.isNotBlank(mSATResultInfo.getSchoolName())
                && StringUtils.isNotBlank(mSATResultInfo.getNumOfSatTestTakers()) && SchoolUtils.isThisInteger(mSATResultInfo.getNumOfSatTestTakers())
                && StringUtils.isNotBlank(mSATResultInfo.getSatCriticalReadingAvgScore()) && SchoolUtils.isThisInteger(mSATResultInfo.getSatCriticalReadingAvgScore())
                && StringUtils.isNotBlank(mSATResultInfo.getSatMathAvgScore()) && SchoolUtils.isThisInteger(mSATResultInfo.getSatMathAvgScore())
                && StringUtils.isNotBlank(mSATResultInfo.getSatWritingAvgScore()) && SchoolUtils.isThisInteger(mSATResultInfo.getSatWritingAvgScore())
                ) {
            return true;
        }
        return false;
    }

}

File: app/src/main/java/com/gcoders/schoolinfo/schoolsfindersystem/view/SchoolMainActivity.kt
package com.gcoders.schoolinfo.schoolsfindersystem.view

import android.content.Intent
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.gcoders.schoolinfo.schoolsfindersystem.R
import com.gcoders.schoolinfo.schoolsfindersystem.SchoolApplication
import com.gcoders.schoolinfo.schoolsfindersystem.model.SchoolListInfo
import com.gcoders.schoolinfo.schoolsfindersystem.model.SchoolSATResultInfo
import com.gcoders.schoolinfo.schoolsfindersystem.presenter.SchoolListPresenter
import com.gcoders.schoolinfo.schoolsfindersystem.utils.SchoolUtils
import com.gcoders.schoolinfo.schoolsfindersystem.view.adapter.SchoolListAdapter
import org.apache.commons.lang3.StringUtils
import java.util.*

class SchoolMainActivity : AppCompatActivity(), SchoolListInfoViewContract.View {

    private var recyclerview_schools_list: RecyclerView? = null
    private var schoolListPresenter: SchoolListInfoViewContract.Presenter? = null
    private var schoolListInfoObjects: ArrayList<SchoolListInfo>? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)

        schoolListInfoObjects = SchoolApplication.getsInstance()!!.schoolListInfoObjects

        schoolListPresenter = SchoolListPresenter(this)

        recyclerview_schools_list = findViewById(R.id.recyclerview_schools_list)
        recyclerview_schools_list!!.layoutManager = LinearLayoutManager(this, RecyclerView.VERTICAL, false)
        schoolListReady(schoolListInfoObjects!!)
    }


    override fun schoolListReady(schoolListInfoObjects: ArrayList<SchoolListInfo>) {
        val adapter = SchoolListAdapter(schoolListInfoObjects, object : SchoolListAdapter.SchoolInfoClickListener {
            override fun onSchoolInfoClick(schoolListInfo: SchoolListInfo) {
                schoolListPresenter!!.getSchoolSATResultInfo(schoolListInfo.dbn!!)
            }
        })
        recyclerview_schools_list!!.adapter = adapter
    }

    override fun schoolInfoClicked(schoolSATResultInfos: ArrayList<SchoolSATResultInfo>) {

        if (null != schoolSATResultInfos && schoolSATResultInfos.size > 0) {
            if (validateSATResults(schoolSATResultInfos)) {
                val intent = Intent(this, SchoolSATscoreActivity::class.java)
                val bundle = Bundle()
                bundle.putParcelable("schoolsatscore", schoolSATResultInfos[0])
                intent.putExtras(bundle)
                startActivity(intent)
            } else {
                onFailure("Incorrect info received. Please try again later..!")
            }
        } else {
            onFailure("No Results found. Please try a different search..!")
        }
    }

    override fun onFailure(errorMessage: String) {
        Toast.makeText(this, errorMessage, Toast.LENGTH_SHORT).show()
    }

    private fun validateSATResults(schoolSATResultInfos: ArrayList<SchoolSATResultInfo>): Boolean {
        val mSATResultInfo = schoolSATResultInfos[0]
        return if (StringUtils.isNotBlank(mSATResultInfo.dbn)
                && StringUtils.isNotBlank(mSATResultInfo.schoolName)
                && StringUtils.isNotBlank(mSATResultInfo.numOfSatTestTakers) && SchoolUtils.isThisInteger(mSATResultInfo.numOfSatTestTakers!!)
                && StringUtils.isNotBlank(mSATResultInfo.satCriticalReadingAvgScore) && SchoolUtils.isThisInteger(mSATResultInfo.satCriticalReadingAvgScore!!)
                && StringUtils.isNotBlank(mSATResultInfo.satMathAvgScore) && SchoolUtils.isThisInteger(mSATResultInfo.satMathAvgScore!!)
                && StringUtils.isNotBlank(mSATResultInfo.satWritingAvgScore) && SchoolUtils.isThisInteger(mSATResultInfo.satWritingAvgScore!!)) {
            true
        } else false
    }

}
